### DISCUSSION TITLE
Is there a HA solution use embed store without SQL DB

### DISCUSSION THREAD

**Discussion by yufeiminds** [polarismesh/polaris#305] [General]
Currently, according to `polaris` deployment model, the data is stored into RDBMS via SQL, so the HA solution is dependent on the database. There is extra available complexity.

Should we provide a store that embeds in `polaris`? let the store and registry all in one, like `consul` and `nacos`.

Hope to have a discussion with you.

Relates links:

- [Consul multi-cluster federation](https://www.consul.io/docs/k8s/installation/multi-cluster)

**Comment by chuntaojun** [comment]
Polaris' storage is plug-in, which can provide an embedded distributed storage plug-in, but this will lose the ability of rapid horizontal expansion, and I think the storage can completely use the existing storage components, such as TiKV can be used in KV , RDBMS can also use NewSQL, built-in distributed storage, reduces operation and maintenance costs, but adds an understanding of polaris built-in distributed storage and operation and maintenance
